src/gateway/shard.rs

```diff
@@ -442,7 +425,7 @@ impl Shard {
     pub fn handle_event(&mut self, event: &Result<GatewayEvent>) -> Result<Option<ShardAction>> {
         match event {
             Ok(GatewayEvent::Dispatch(seq, event)) => Ok(self.handle_gateway_dispatch(*seq, event)),
-            Ok(GatewayEvent::Heartbeat(s)) => Ok(Some(self.handle_heartbeat_event(*s))),
+            Ok(GatewayEvent::Heartbeat(..)) => Ok(Some(self.handle_heartbeat_event())),
```

Comment:
You could probably inline the implementation of `handle_heartbeat_event` here, now that it's only a few lines.